benchmark_test.go

```diff
@@ -609,3 +619,217 @@ func Benchmark_S3_segupload(b *testing.B) {
+
+	_, err = writer.Write(data)
+	if err != nil {
+		panic(err)
+	}
+
+	err = writer.Flush()
+	if err != nil {
+		panic(err)
+	}
+}
+
+func benchmarkCuckooPanmari(strs [][]byte) {
+	// Default config: https://github.com/panmari/cuckoofilter
+	// b (bucket size): 4
+	// f (fingerprint): 16
+	// False positive rate: 0.0001
+	numKeys := uint(len(strs)) // num of keys the filter will store
+
+	cuckoo := cuckooPan.NewFilter(numKeys)
+
+	for _, str := range strs {
+		cuckoo.Insert(str) // Also try insert unique
+	}
+
+	fname := "cuckoo_filter_pan.bin"
+	file, err := os.OpenFile(fname, os.O_WRONLY|os.O_CREATE, 0644)
+	if err != nil {
+		panic(err)
+	}
+	defer file.Close()
+
+	writer := bufio.NewWriter(file)
+
+	data := cuckoo.Encode()
+
+	_, err = writer.Write(data)
+	if err != nil {
+		panic(err)
+	}
+
+	err = writer.Flush()
+	if err != nil {
+		panic(err)
+	}
+}
+
+// // Function to deserialize binaryfuse8 filter from a file
+// func deserializeBinaryFuse8Filter(filename string) (*xorfilter.BinaryFuse8, error) {
+//     file, err := os.Open(filename)
+//     if err != nil {
+//         return nil, err
+//     }
+//     defer file.Close()
+
+//     var filter xorfilter.BinaryFuse8
+//     decoder := gob.NewDecoder(file)
+//     err = decoder.Decode(&filter)
+//     return &filter, err
+// }
+
+// // Function to deserialize xor8 filter from a file
+// func deserializeXor8Filter(filename string) (*xorfilter.Xor8, error) {
+//     file, err := os.Open(filename)
+//     if err != nil {
+//         return nil, err
+//     }
+//     defer file.Close()
+
+//     var filter xorfilter.Xor8
+//     decoder := gob.NewDecoder(file)
+//     err = decoder.Decode(&filter)
+//     return &filter, err
+// }
+
+func benchmarkXorFilter(hashedKeys []uint64) {
+	// Configs: https://github.com/FastFilter/xorfilter
+
+	xorfilter, err := xorfilter.Populate(hashedKeys) // Use PopulateBinaryFuse8 for binary fuse filter
+	if err != nil {
+		panic(err)
+	}
+
+	fname := "xor_filter.bin"
+	file, err := os.OpenFile(fname, os.O_WRONLY|os.O_CREATE, 0644)
+	if err != nil {
+		panic(err)
+	}
+	defer file.Close()
+
+	encoder := gob.NewEncoder(file)
+	err = encoder.Encode(xorfilter)
+	if err != nil {
+		panic(err)
+	}
+
+}
+
+func Benchmark_Filters(b *testing.B) {
+	N := 10_000_000
+
+	randomStrs := make([][]byte, N)
+	hashedKeys := make([]uint64, N)
+	for i := 0; i < N; i++ {
+		randomStrs[i] = []byte(putils.GetRandomString(100, putils.AlphaNumeric))
+		hashedKeys[i] = xxhash.Sum64(randomStrs[i])
+	}
+
+	benchmarkBloom(randomStrs)
+	benchmarkCuckooLinvon(randomStrs)
+	benchmarkCuckooSeiflotfy(randomStrs)
+	benchmarkCuckooPanmari(randomStrs)
+	benchmarkXorFilter(hashedKeys)
+
+	/*
+	   go test -run=Bench -bench=Benchmark_Filters  -cpuprofile cpuprofile.out -o rawsearch_cpu
+	   go tool pprof ./rawsearch_cpu
+
+	   (for mem profile)
+	   go test -run=Bench -bench=Benchmark_Filters -benchmem -memprofile memprofile.out -o rawsearch_mem
+	   go tool pprof ./rawsearch_mem memprofile.out
+	*/
```

Comment:
nit: move this comment near the top of the function

Reply:
addressed in the latest commit.